src/modules/browser.ts

```diff
@@ -24,25 +24,31 @@ export class Browser {
     return this.browser!;
   }
   async takeScreenshot(url: string, selector: string, viewport: ViewPort) {
-    const start = new Date().getTime();
-    const browser = await this.getBrowser();
-    const page = await browser.newPage();
-    await page.setViewport({
-      deviceScaleFactor: 2,
-      ...viewport,
-    });
-    await page.goto(url);
-    const container = await page.waitForSelector(selector);
-    if (!container) {
-      throw new Error(`Could not generate screenshot`);
+    try {
+      const browser = await this.getBrowser();
+      const page = await browser.newPage();
+      await page.setViewport({
+        deviceScaleFactor: 2,
+        ...viewport,
+      });
+      await page.goto(url);
+      const container = await page.waitForSelector(selector);
+      if (!container) {
+        throw new Error(`Could not generate screenshot`);
+      }
+      const buffer = await container.screenshot({
+        encoding: "binary",
+        omitBackground: true,
+      });
+      await page.close();
+      return buffer as Buffer;
+    } catch (error) {
+      this.reset();
+      throw error;
     }
-    const buffer = await container.screenshot({
-      encoding: "binary",
-      omitBackground: true,
-    });
-    await page.close();
-    const end = new Date().getTime();
-    console.log(`Screenshot for ${url} took ${end - start}ms`)
-    return buffer as Buffer;
+  }
+
+  reset() {
+    delete this.browser;
```

Comment:
Shouldn't we attempt to close the underlying puppeteer browser first?

Reply:
good catch! I added it